src/twisted/python/bytes.py

```diff
@@ -0,0 +1,38 @@
+# -*- test-case-name: twisted.python.test.test_bytes -*-
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+
+"""
+Utility for converting between L{unicode} and L{bytes}
+"""
+
+from __future__ import division, absolute_import, print_function
+
+from twisted.python.compat import unicode
+
+
+def ensureBytes(s, encoding="ascii", errors="strict"):
+    """
+    Convert L{unicode} string to L{bytes}.
+
+    @param s: The string to convert.
+    @type s: L{unicode} or L{bytes}
+    @param encoding: The encoding to pass to L{str.encode}.
+    @param errors: The error handling scheme to pass to L{str.encode}.
+
+    @raise TypeError: The input is not L{unicode} or L{bytes}.
+
+    @return: The encoded string.  If I{s} is L{bytes} just return I{s}.
+    @rtype: L{bytes}
+    """
+    if isinstance(s, bytes):
+        return s
+    elif isinstance(s, unicode):
+        return s.encode(encoding=encoding, errors=errors)
+    else:
+        raise TypeError("Expected {} to be unicode or bytes not: "
```

Comment:
Not sure if we need this here.
It is documented that it needs Unicode or bytes. When people pass something else, they are on their own.

Reply:
@adiroiban
are you OK if I change `ensureBytes` to look like https://github.com/lyft/syx/blob/master/syx/__init__.py#L23 with no `raise TypeError`?

Also, I think I want to move this function inside `twisted.python.__init__.py` instead
of having it in a separate bytes.py file.  Are you OK with that?

Reply:
No.  No implementations in `__init__.py`.  And `twisted.python` doesn't expose an API except for the modules it containers.

Reply:
I am ok with the implementation from syx.
I am +1 for not raising TypeError and is ok to add an encoding mode (with strict as default)... but I guess that we can add the argument later if required.
For now we can document that it goes with strict encoding.

-1 for moving in `__ini__ `. We can keep it in a separate module, as is now in the PR

Thanks